appium/webdriver/webdriver.py

```diff
@@ -595,9 +595,27 @@ def battery_info(self):
 
     # pylint: disable=protected-access
 
+    def _get_webdriver_parents(self, mro):
+        """
+        Get Method Resolution Order after `appium.webdriver.webdriver.WebDriver`
+        to avoid __addCommands loop in `appium.webdriver.webdriver.WebDriver`
+
+        :Args:
+            - Tuple of Method Resolution Order
+        :Raises:
+            - ValueError: If the MRO has no 'appium.webdriver.webdriver.WebDriver',
+            it raises 'tuple.index(x): x not in tuple' error
+        """
+        m = tuple(map(lambda x: '{}.{}'.format(x.__module__, x.__name__), mro))
+        wd_index = m.index('{}.WebDriver'.format(__name__))
+
+        return mro[(wd_index + 1):]
+
     def _addCommands(self):
-        # call the overridden command binders from all mixin classes
-        for mixin_class in filter(lambda x: x is not self.__class__, self.__class__.__mro__):
+        # call the overridden command binders from all mixin classes except for
+        # appium.webdriver.webdriver.WebDriver and its sub-classes
+        # https://github.com/appium/python-client/issues/342
+        for mixin_class in self._get_webdriver_parents(self.__class__.__mro__):
```

Comment:
`filter(lambda x: not issubclass(x, self.__class__), self.__class__.__mro__)` should be fine

Reply:
Yes! I could not find it...thanks so much